HEDNS fails HTTP/ech= test

**Describe the bug**
HEDNS fails the `testgroup("Ech",` test.

**To Reproduce**
```
$ go test -run TestDNSProviders -timeout 50m -v -verbose -profile HEDNS -start 18 -end 18
=== RUN   TestDNSProviders
Testing Profile="HEDNS" (TYPE="HEDNS")
=== RUN   TestDNSProviders/dnscontrol.net
=== RUN   TestDNSProviders/dnscontrol.net/Clean_Slate:Empty
    helpers_integration_test.go:236: 
        - DELETE dnscontrol.net HTTPS 1 example.com. alpn=h2,h3 ttl=300
=== RUN   TestDNSProviders/dnscontrol.net/18:Ech:Create_a_HTTPS_record
    helpers_integration_test.go:236: 
        + CREATE dnscontrol.net HTTPS 1 example.com. alpn=h2,h3 ttl=300
=== RUN   TestDNSProviders/dnscontrol.net/18:Ech:Add_an_ECH_key
    helpers_integration_test.go:236: 
        ± MODIFY dnscontrol.net HTTPS (1 example.com. alpn=h2,h3 ttl=300) -> (1 example.com. alpn=h2,h3 ech=some+base64+encoded+value/// ttl=300)
=== RUN   TestDNSProviders/dnscontrol.net/18:Ech:Ignore_the_ECH_key_while_changing_other_values
    helpers_integration_test.go:236: 
        ± MODIFY dnscontrol.net HTTPS (1 example.com. alpn=h2,h3 ech=some+base64+encoded+value/// ttl=300) -> (1 example.net. port=80 ech="some+base64+encoded+value///" ttl=300)
    helpers_integration_test.go:257: Expected 0 corrections on second run, but found 1.
    helpers_integration_test.go:259: UNEXPECTED #0: ± MODIFY dnscontrol.net HTTPS (1 example.net. port=80 ech=some+base64+encoded+value/// ttl=300) -> (1 example.net. port=80 ech="some+base64+encoded+value///" ttl=300)
--- FAIL: TestDNSProviders (34.46s)
    --- FAIL: TestDNSProviders/dnscontrol.net (32.37s)
        --- PASS: TestDNSProviders/dnscontrol.net/Clean_Slate:Empty (5.44s)
        --- PASS: TestDNSProviders/dnscontrol.net/18:Ech:Create_a_HTTPS_record (9.37s)
        --- PASS: TestDNSProviders/dnscontrol.net/18:Ech:Add_an_ECH_key (8.95s)
        --- FAIL: TestDNSProviders/dnscontrol.net/18:Ech:Ignore_the_ECH_key_while_changing_other_values (8.60s)
FAIL
exit status 1
FAIL	github.com/DNSControl/dnscontrol/v4/integrationTest	34.951s
```

**Expected behavior**
The test passes

**DNS Provider**
 - HEDNS

**Additional context**
Add any other context about the problem here.

--- a/.github/workflows/pr_integration_tests.yml
+++ b/.github/workflows/pr_integration_tests.yml
@@ -300,7 +300,7 @@ jobs:
             # NB(tlim): THe output of gotestsum needs to go to stdout AND the
             # gotestsum-output.txt file. That's why "tee" is used.
             gotestsum --junitfile ${TEST_RESULTS}/gotestsum-report.xml -- \
-                -timeout 40m -v \
+                -timeout 70m -v \
                 -verbose \
                 -profile ${{ matrix.provider }} \
                 -cfworkers=true \

--- a/integrationTest/integration_test.go
+++ b/integrationTest/integration_test.go
@@ -317,7 +317,6 @@ func makeTests() []*TestGroup {
 				//
 				// Let's just ignore ECH test for Vercel for now.
 				"VERCEL",
-				"HEDNS", // BUG: https://github.com/DNSControl/dnscontrol/issues/4328
 			),
 			tc("Create a HTTPS record", https("@", 1, "example.com.", "alpn=h2,h3")),
 			tc("Add an ECH key", https("@", 1, "example.com.", "alpn=h2,h3 ech=some+base64+encoded+value///")),

